### PR TITLE
CAN-2168 horovod params description

### PR DESCRIPTION
**Pull request by mateusz-ciesielski**
* Add description to parameters in values.yaml
* Add recommended MKL env variables for increased training performance

### CODE REVIEW

**Comment by mateusz-ciesielski** [comment]
Tests:
https://ci.nervana.sclab.intel.com/job/carbon/job/k8s/job/K8sClusterOps-builder01/371/

**Review by mzylowski — APPROVED** [review]

File: jupyter/charts/values.yaml [hunk 1]
```diff
@@ -2,15 +2,7 @@
...
-JupyterPort: 8888
-
-image:
-  pullPolicy: IfNotPresent
-  clusterRepository: {{ NAUTA.ExperimentImage }}
-
-service:
-  type: ClusterIP
-
+# Resorce settings
```
Comment (jcchr) [inline jupyter/charts/values.yaml]: Resource

File: multinode-tf-training-horovod-py2/charts/values.yaml [hunk 2]
```diff
@@ -2,10 +2,71 @@
...
+master:
+  env:
+    - name: HOROVOD_TIMELINE
+      value: "/mnt/output/experiment/timeline.json"  # Path to where horovod timeline log will be saved
+    - name: OMP_NUM_THREADS
+      value: "20"  # Set to the number of physical cores used by single worker
+    - name: KMP_BLOCKTIME
+      value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+    - name: KMP_AFFINITY
+      value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline multinode-tf-training-horovod-py2/charts/values.yaml]: recommended

File: multinode-tf-training-horovod-py2/charts/values.yaml [hunk 3]
```diff
@@ -2,10 +2,71 @@
...
+
+worker:
+  podManagementPolicy: Parallel
+  env:
+    - name: OMP_NUM_THREADS
+      value: "20"  # Set to the number of physical cores used by single worker
+    - name: KMP_BLOCKTIME
+      value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+    - name: KMP_AFFINITY
+      value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline multinode-tf-training-horovod-py2/charts/values.yaml]: recommend

File: multinode-tf-training-horovod/charts/values.yaml [hunk 4]
```diff
@@ -2,10 +2,71 @@
...
+master:
+  env:
+    - name: HOROVOD_TIMELINE
+      value: "/mnt/output/experiment/timeline.json"  # Path to where horovod timeline log will be saved
+    - name: OMP_NUM_THREADS
+      value: "20"  # Set to the number of physical cores used by single worker
+    - name: KMP_BLOCKTIME
+      value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+    - name: KMP_AFFINITY
+      value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline multinode-tf-training-horovod/charts/values.yaml]: recommended

File: multinode-tf-training-horovod/charts/values.yaml [hunk 5]
```diff
@@ -2,10 +2,71 @@
...
+
+worker:
+  podManagementPolicy: Parallel
+  env:
+    - name: OMP_NUM_THREADS
+      value: "20"  # Set to the number of physical cores used by single worker
+    - name: KMP_BLOCKTIME
+      value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+    - name: KMP_AFFINITY
+      value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline multinode-tf-training-horovod/charts/values.yaml]: recommended

File: multinode-tf-training-tfjob-py2/charts/values.yaml [hunk 6]
```diff
@@ -2,22 +2,7 @@
...
-    {% endfor %}
-
-image:
-  pullPolicy: IfNotPresent
-  clusterRepository: {{ NAUTA.ExperimentImage }}
-
-service:
-  type: ClusterIP
-
+# Resorce settings for training workers
```
Comment (jcchr) [inline multinode-tf-training-tfjob-py2/charts/values.yaml]: resource

File: multinode-tf-training-tfjob-py2/charts/values.yaml [hunk 7]
```diff
@@ -48,14 +37,45 @@ sidecar_resources:
...
-experimentName: {{ NAUTA.ExperimentName }} 
-registryPort: {{ NAUTA.RegistryPort }}
+# Environment variables that will be set on training container
+env:
+  - name: OMP_NUM_THREADS
+    value: "20"  # Set to the number of physical cores used by single worker
+  - name: KMP_BLOCKTIME
+    value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+  - name: KMP_AFFINITY
+    value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline multinode-tf-training-tfjob-py2/charts/values.yaml]: recommended

File: multinode-tf-training-tfjob/charts/values.yaml [hunk 8]
```diff
@@ -2,22 +2,7 @@
...
-    {% endfor %}
-
-image:
-  pullPolicy: IfNotPresent
-  clusterRepository: {{ NAUTA.ExperimentImage }}
-
-service:
-  type: ClusterIP
-
+# Resorce settings for training workers
```
Comment (jcchr) [inline multinode-tf-training-tfjob/charts/values.yaml]: resource

File: multinode-tf-training-tfjob/charts/values.yaml [hunk 9]
```diff
@@ -26,9 +11,11 @@ worker_resources:
     cpu: 76
     memory: 252.7Gi
 
+# Shorthands for setting both limits and requests of workers' cpu/memory
 worker_cpu: null
 worker_memory: null
 
+# Resorce settings for parameter servers
```
Comment (jcchr) [inline multinode-tf-training-tfjob/charts/values.yaml]: resource

File: multinode-tf-training-tfjob/charts/values.yaml [hunk 10]
```diff
@@ -48,16 +37,45 @@ sidecar_resources:
...
 sidecar_memory: null
 
+# Environment variables that will be set on training container
+env:
+  - name: OMP_NUM_THREADS
+    value: "20"  # Set to the number of physical cores used by single worker
+  - name: KMP_BLOCKTIME
+    value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+  - name: KMP_AFFINITY
+    value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline multinode-tf-training-tfjob/charts/values.yaml]: recommended

File: pytorch/charts/values.yaml [hunk 11]
```diff
@@ -2,16 +2,7 @@
...
-  args:
-    {% for arg in NAUTA.CommandLine %}
-    - {{ arg }}
-    {% endfor %}
-
-image:
-  pullPolicy: IfNotPresent
-  clusterRepository: {{ NAUTA.ExperimentImage }}
-
+# Resorce settings for training workers
```
Comment (jcchr) [inline pytorch/charts/values.yaml]: resource

File: pytorch/charts/values.yaml [hunk 12]
```diff
@@ -20,9 +11,11 @@ worker_resources:
     cpu: 76
     memory: 252.7Gi
 
+# Shorthands for setting both limits and requests of workers' cpu/memory
 worker_cpu: null
 worker_memory: null
 
+# Resorce settings for master
```
Comment (jcchr) [inline pytorch/charts/values.yaml]: resource

File: tf-inference-batch/charts/values.yaml [hunk 13]
```diff
@@ -2,20 +2,10 @@
...
-containerPort: 8500
-
-image:
-  pullPolicy: IfNotPresent
-  clusterRepository: {{ NAUTA.ExperimentImage }}
-
+# Count of batch inference workers
 workersCount: 1
 
+# Resorce settings for batch inference instance
```
Comment (jcchr) [inline tf-inference-batch/charts/values.yaml]: resource

File: tf-inference-batch/charts/values.yaml [hunk 14]
```diff
@@ -24,22 +14,40 @@ resources:
     cpu: 38
     memory: 126.35Gi
 
+# Shorthands for setting both limits and requests of batch inference instance's cpu/memory
 cpu: null
 memory: null
 
+# Resorce settings for batch inference wrapper
```
Comment (jcchr) [inline tf-inference-batch/charts/values.yaml]: resource

File: tf-inference-stream/charts/values.yaml [hunk 15]
```diff
@@ -2,18 +2,10 @@
...
-grpcContainerPort: 8500
-
-image:
-  pullPolicy: IfNotPresent
-  clusterRepository: {{ NAUTA.ExperimentImage }}
-
+# Count of stream inference workers
 workersCount: 1
 
+# Resorce settings for stream inference instance
```
Comment (jcchr) [inline tf-inference-stream/charts/values.yaml]: resource

File: tf-training-tfjob-py2/charts/values.yaml [hunk 16]
```diff
@@ -2,22 +2,7 @@
...
-    {% endfor %}
-
-image:
-  pullPolicy: IfNotPresent
-  clusterRepository: {{ NAUTA.ExperimentImage }}
-
-service:
-  type: ClusterIP
-
+# Resorce settings for training pod
```
Comment (jcchr) [inline tf-training-tfjob-py2/charts/values.yaml]: resource

File: tf-training-tfjob-py2/charts/values.yaml [hunk 17]
```diff
@@ -26,10 +11,39 @@ resources:
...
 cpu: null
 memory: null
 
+env:
+  - name: OMP_NUM_THREADS
+    value: "20"  # Set to the number of physical cores used by single worker
+  - name: KMP_BLOCKTIME
+    value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+  - name: KMP_AFFINITY
+    value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline tf-training-tfjob-py2/charts/values.yaml]: recommend

File: tf-training-tfjob/charts/values.yaml [hunk 18]
```diff
@@ -26,11 +11,39 @@ resources:
...
 
-experimentName: {{ NAUTA.ExperimentName }} 
-podCount: 1
+env:
+  - name: OMP_NUM_THREADS
+    value: "20"  # Set to the number of physical cores used by single worker
+  - name: KMP_BLOCKTIME
+    value: "0"  # 0 is a good default for most CNN models, could be adjusted for additional performance
+  - name: KMP_AFFINITY
+    value: "granularity=fine,verbose,compact,1,0"  # these are reccomended affinity settings
```
Comment (jcchr) [inline tf-training-tfjob/charts/values.yaml]: recommend

**Review by jcchr — CHANGES_REQUESTED** [review]

**Comment by mateusz-ciesielski** [comment]
@jcchr Typos fixed

File: tf-training-tfjob-py2/charts/values.yaml [hunk 19]
```diff
@@ -26,10 +11,39 @@ resources:
     cpu: 19
     memory: 93604421089
 
+# Shorthands for setting both limits and requests of cpu/memory
 cpu: null
 memory: null
 
+env:
+  - name: OMP_NUM_THREADS
```
Comment (jcchr) [inline tf-training-tfjob-py2/charts/values.yaml:19]: tf-training-tfjob is not a multinode training - so it doesn't use OpenMPI - thus those variables are superfluous here.

File: tf-training-tfjob/charts/values.yaml [hunk 20]
```diff
@@ -26,11 +11,39 @@ resources:
...
     memory: 93604421089
 
+# Shorthands for setting both limits and requests of cpu/memory
 cpu: null
 memory: null
 
-experimentName: {{ NAUTA.ExperimentName }} 
-podCount: 1
+env:
+  - name: OMP_NUM_THREADS
```
Comment (jcchr) [inline tf-training-tfjob/charts/values.yaml:19]: tf-training-tfjob is not a multinode training - so it doesn't use OpenMPI - thus those variables are superfluous here.

File: multinode-tf-training-tfjob-py2/charts/values.yaml [hunk 21]
```diff
@@ -48,14 +37,45 @@ sidecar_resources:
...
 
+# Shorthands for setting both limits and requests of sidecar's cpu/memory
 sidecar_cpu: null
 sidecar_memory: null
 
-experimentName: {{ NAUTA.ExperimentName }} 
-registryPort: {{ NAUTA.RegistryPort }}
+# Environment variables that will be set on training container
+env:
+  - name: OMP_NUM_THREADS
```
Comment (jcchr) [inline multinode-tf-training-tfjob-py2/charts/values.yaml:46]: In this pack we don't use OpenMPI - so those variables are not needed.

File: multinode-tf-training-tfjob/charts/values.yaml [hunk 22]
```diff
@@ -48,16 +37,45 @@ sidecar_resources:
     cpu: 0.01
     memory: 100Mi
 
+# Shorthands for setting both limits and requests of sidecar's cpu/memory
 sidecar_cpu: null
 sidecar_memory: null
 
+# Environment variables that will be set on training container
+env:
+  - name: OMP_NUM_THREADS
```
Comment (jcchr) [inline multinode-tf-training-tfjob/charts/values.yaml:46]: In this pack we don't use OpenMPI - so those variables are not needed.

**Review by jcchr — CHANGES_REQUESTED** [review]

**Comment by mateusz-ciesielski** [comment]
@jcchr 
According to https://www.tensorflow.org/guide/performance/overview#tuning_mkl_for_the_best_performance `OMP_NUM_THREADS` is used by MKL (so not only OpenMPI) for performance tuning

**Review by jcchr — APPROVED** [review]

**Comment by mateusz-ciesielski** [comment]
Tests:
https://ci.nervana.sclab.intel.com/job/carbon/job/k8s/job/K8sClusterOps/13632/

**Pull request merged**